src/common/imagpng.cpp

```diff
@@ -256,7 +256,9 @@ wxPNGHandler::LoadFile(wxImage *image,
     // VZ: as this function uses setjmp() the only fool-proof error handling
     //     method is to use goto (setjmp is not really C++ dtors friendly...)
 
-    unsigned char **lines = NULL;
+    // allocate the lines pointer dynamically to avoid clobbering by longjmp
+    unsigned char *** const lines_p = new unsigned char ** (NULL);
+    unsigned char ** &lines = *lines_p;
```

Comment:
Sorry, I don't understand what's going on here and how does this actually help. What happens with `lines_p` after `longjmp()`?

Reply:
Local variable's value is indeterminate after longjmp, if it was modified after setjmp (https://en.cppreference.com/w/cpp/utility/program/setjmp). In practice this means that setjmp/longjmp can save/restore registers even if they are used to store local variables, so longjmp could reset lines to null and cause a memory leak. lines_p is never modified, so longjmp will just overwrite it with the same value.

Reply:
After rereading this once again, I realize that you probably rely on `lines_p` being unchanged because it is `const`. But I'm not actually sure if this really works like this... Are you?

I think it would be safer to have a helper function and pass it a pointer to `lines` instead. Because even if this code actually works, it's rather mysterious and I'd like to find a simpler way of fixing the problem.